src/mechanics/timeIntegration/TimeControl.cpp

```diff
@@ -0,0 +1,109 @@
+#include "TimeControl.h"
+
+
+
+void NuTo::TimeControl::Proceed()
+{
+    if (mTimeStep <= 0.0)
+        throw MechanicsException(__PRETTY_FUNCTION__,"Current timestep is 0 or negative!");
+    mPreviousTime = mCurrentTime;
+    mCurrentTime += mTimeStep;
+//    if (mCurrentTime > mTimeFinal)
+//    {
+//        mCurrentTime = mTimeFinal;
+//        mTimeStep = mTimeFinal - mCurrentTime;
+//    }
+}
+
+void NuTo::TimeControl::AdjustTimestep(int iterations, int maxIterations, bool converged)
+{
+
+
+    mTimeStep = mTimeStepFunction(*this,iterations,maxIterations,converged);
+
+    if(mTimeStep > mMaxTimeStep)
+        mTimeStep =  mMaxTimeStep;
+
+    if(!converged && mPreviousTime !=mCurrentTime)
+        throw MechanicsException(__PRETTY_FUNCTION__, "No convergence with the current maximum number of "
+                                                      "iterations, either use automatic time stepping, "
+                                                      "reduce the time step or the minimal line search cut "
+                                                      "back factor. In case you provided a custom timestepping function "
+                                                      "and intend to use some kind of automatic timestepping, call "
+                                                      "the RestorePreviosTime() function of the time control before reducing "
+                                                      "the timestep.");
+}
+
+
+
+void NuTo::TimeControl::SetTimeStep(double timeStep)
+{
+    if (timeStep<=0)
+        throw MechanicsException(__PRETTY_FUNCTION__,"Timestep must be larger tham 0!");
+
+    mTimeStep = timeStep;
+}
+
+void NuTo::TimeControl::SetTimeFinal(double timeFinal)
+{
+    if (timeFinal <= mCurrentTime)
+    {
+        throw MechanicsException(__PRETTY_FUNCTION__,"Final time must be larger than current time!");
+    }
+    mTimeFinal          = timeFinal;
```

Comment:
Way too many spaces here... Please setup clang-format and run it! Merci much! (Then, IMO, this can be merged)